app/controllers/api/v2/registration_commands_controller.rb

```diff
@@ -15,6 +15,8 @@ class RegistrationCommandsController < V2::BaseController
         param :setup_remote_execution, :bool, desc: N_("Set 'host_registration_remote_execution' parameter for the host. If it is set to true, SSH keys will be installed on the host")
         param :jwt_expiration, :number, desc: N_("Expiration of the authorization token (in hours)")
         param :insecure, :bool, desc: N_("Enable insecure argument for the initial curl")
+        param :repo, String, desc: N_("Repository URL / details, for example for Debian OS family: 'deb http://deb.example.com/ buster 1.0', for Red Hat OS family: 'http://rpm.example.com/'")
+        param :repo_gpg, String, desc: N_("GPG key for the repository")
```

Comment:
The code implies this must be a URL (since curl is used). I wonder if users might actually post the GPG key so they don't rely on network. I'd at least mention that this parameter expects URLs.

Does Katello have GPG keys available as inline data? If so, I'd consider making registration recognize http(s):// links and handle inline data.

Reply:
Description updated.

> Does Katello have GPG keys available as inline data?
Not sure if I understand, what do you mean by inline data?

Reply:
Katello allows to upload GPG keys as a Content Credential objects, they [accept the actual content](https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/content_credentials_controller.rb#L15) of the key. When users would like to use custom Katello repository, they would need to first download the GPG key from Katello and put it to some web server to host it.

It would be nice if this parameter accepted the entire public key or the url and detect that. Not sure if we need to mix it into this PR though. The problem is, if we introduce this param we shouldn't change it's meaning later.

Reply:
My preference: if it's feasible before 2.5 stabilization, I'd like to this parameter to accept also the entire pub key, if not, I'm fine with just renaming this param to repo_gpg_key_url and introducing the repo_gpg_key as a separate parameter at some point later.

Reply:
What @ares said. I would even consider *only* accepting the full GPG key and no URLs. It is arguably more secure and IMHO not a big overhead for users. Additionally, it's one less step that could fail during registration due to broken networking.

Reply:
Do we really want to pass whole key as URL parameter? The final registration command would be **insanely** long & totally user-unfriendly IMO.

Bootstrap.py [0] accepts `gpg_key`  path to file, so we can do it the same way, if getting the key from URL is not good option.

[0] https://github.com/Katello/katello-client-bootstrap#providing-a-repository-with-the-subscription-manager-packages

@ares @ekohl opinions?

Reply:
My reasoning is that if I'm a Katello user that when I see `repo_gpg` without context, I'll assume that it's the full key. That's why @ares suggested to rename it to `repo_gpg_key_url`. It would be clear on the user's side what the expected data is. A help text is one thing, but a self documenting parameter is even better.

Another thought I had is that a hostgroup might already have content repositories assigned and reuse those, but perhaps that's irrelevant and a key URL would be sufficient.

IMHO the goal is to have a parameter that's hard to misuse to minimize potential user frustration. I'm less concerned about the actual feature set behind it right now. Does that help?